// Protos that the patch+subset incremental transfer client and server use to
// communicate.
//
// See Patch+Subset incremental transfer design doc for more info:
// go/patch-subset-design-doc
syntax = "proto3";

package patch_subset;

message PatchResponseProto {
  int32 format = 1;
  bytes patch = 2;
  bytes replacement = 3;
  uint64 original_font_checksum = 4;
  uint64 patched_checksum = 5;
  CompressedListProto codepoint_ordering = 6;
  uint64 ordering_checksum = 7;
}

message CompressedListProto {
  // The value at index, i is deltas[i-1] + deltas[i].
  // deltas[-1] is assumed to be 0.
  repeated sint32 deltas = 2;
}

